src/inverted-index.js

```diff
@@ -0,0 +1,200 @@
+    let indexDict = {};
+    //loop through the documents
+    for(let each in documents) {
+      let tokenArray = documents[each].textTokens;
+      let tokenLength = tokenArray.length;
+      for(let i = 0; i < tokenLength; i++) {
+        let token = tokenArray[i];
+        //check if the word has not been indexed and used as key in the object
+        if(!hasProperty.call(indexDict, token)) {
+          //the token is used as a key and initialized to an empty array
+          indexDict[token] = [];
+        }
+        //a check is run to confirm if the document has been indexed
+        //with the word
+        if(indexDict[token].indexOf(documents[each].documentNum) === -1) {
+          indexDict[token].push(documents[each].documentNum);
+        }
+      }
+    }
+    return indexDict;
+  }
+
+  /**
+  getIndex method returns the indexed words and the documents they were found
+  @filename {string} name of the file to get its index
+  @return {Object} the words index
+  */
+  getIndex(filename) {
+    let returnValue = this.filesIndexed[filename].index === undefined ? false : this.filesIndexed[filename].index;
+    return returnValue;
+  }
+
+  /**
+  searchIndex searches the indexed words to determine the documents that the
+  searchterms can be found
+  @params searchTerm {string, array} the search query
+  @params filename {string}- the name of the file to search its index
+  @return {object|boolean} it returns boolean if the searchTerm is empty and
+  it retrns object if it is not. Each index is each searcykeyword.
+  Each with an array value of the document index
+  */
+  searchIndex(searchTerm, filename) {
+    if((typeof searchTerm === "string" && searchTerm.trim() === "") ||
+      (typeof searchTerm === "object" && searchTerm.length === 0) ||
+      searchTerm === undefined) {
+        return false;
+    }
+
+    let result = [];
+    if(filename === "all") {
+      for(let eachFile in this.filesIndexed) {
+        result.push({
+          indexes: this.getSearchResults(searchTerm, eachFile),
+          searchedFile: eachFile,
+          documents: this.getDocuments(eachFile)
+        });
+      }
+    } else {
+      result.push({
+        indexes: this.getSearchResults(searchTerm, filename),
+        searchedFile: filename,
+        documents: this.getDocuments(filename)
+      });
+    }
+    return result;
+  }
+
+  /**
+  getSearchResults method checks the index of the file and
+  returns the result
+  @param searchTokens {string or array} - the search query can be a string
+  or an array
+  @param filename {string} - the name of the file
+  @return {object} - an object with the found words as keys
+  */
+  getSearchResults(searchTokens, filename) {
+    let indexToSearch = this.getIndex(filename), result = {};
+    //if it is a string of search terms then a split can be done
+    if(typeof searchTokens === "string") {
+      searchTokens = this.tokenize(searchTokens);
+    }
+    for(let indexCount in searchTokens) {
```

Comment:
This easily has a complexity of O(n**2). It kinds of defeats the purpose of an Inverted Index.